github/messages.go

```diff
@@ -132,17 +137,37 @@ func messageMAC(signature string) ([]byte, func() hash.Hash, error) {
 //       // Process payload...
 //     }
 //
-func ValidatePayload(r *http.Request, secretKey []byte) (payload []byte, err error) {
-	payload, err = ioutil.ReadAll(r.Body)
-	if err != nil {
-		return nil, err
+func ValidatePayload(r *http.Request, secretKey []byte) ([]byte, error) {
+	// jsonPayload is the json webhook data
+	var jsonPayload []byte
+	// rawPayload is the raw body that GitHub uses to calculate the signature
+	var rawPayload []byte
+
+	if r.Header.Get("Content-Type") == "application/x-www-form-urlencoded" {
```

Comment:
Totally agree. FYI though that this will be a backwards incompatible change since previously we never checked the content-type header and now we're rejecting anything without `application/json`. If people are using this as expected then GitHub would always send that header, however it is technically breaking BC.